libs/metrics-wai/src/Data/Metrics/Servant.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
+
+-- | Given a servant API type, this module gives you a 'Paths' for 'withPathTemplate'.
+module Data.Metrics.Servant where
+
+import Imports
+import Data.Metrics.Types
+import Data.Proxy
+import Data.String.Conversions
+import Data.Tree
+import GHC.TypeLits
+import Servant.API
+
+
+routesToPaths :: forall routes. RoutesToPaths routes => Paths
+routesToPaths = Paths (getRoutes @routes)
+
+class RoutesToPaths routes where
+  getRoutes :: Forest (Maybe ByteString)
+
+
+-- "seg" :> routes
+instance {-# OVERLAPPING #-}
```

Comment:
I don't think you'd need both the `OVERLAPPING` and `OVERLAPPABLE`; but I wanted to bring up that I think it's nice that you added it, makes it easier to track down weird behaviour in the future if it's clear that there are additional instances hidden elsewhere :+1: